plugins/rss.ts

```diff
@@ -0,0 +1,90 @@
+import { $XML, stringify } from "../deps/xml.ts";
+import { Page } from "../core/filesystem.ts";
+import { Search } from "../plugins/search.ts";
+
+import type { Data, Site } from "../core.ts";
+
+export interface Options {
+  filename: string;
+  query: string;
+  sort: string;
+  link: string;
+  title: string;
+  buildDate: Date;
+  description: string;
+  language: string;
+}
+
+export const defaults: Options = {
+  filename: "/feed.rss",
+  query: "",
+  sort: "url=asc",
+  link: "http://127.0.0.1:3000",
+  title: "My RSS Feed",
+  buildDate: new Date(),
+  description: "",
+  language: "en",
+};
+
+export default (userOptions?: Partial<Options>) => {
+  const options = {
+    ...defaults,
+    ...userOptions,
+  };
+
+  return (site: Site) => {
+    site.addEventListener("afterRender", () => {
+      const feed = Page.create(options.filename, getFeedContent(site));
+      site.pages.push(feed);
+    });
+
+    const getFeedContent = (site: Site) => {
+      const search = new Search(site, true);
+      const pages = search.pages(options.query, options.sort);
+      const items = pages.map((page: Data) => ({
+        title: page.title,
+        link: `${options.link}${page.url}`,
+        guid: {
+          "@isPermaLink": false,
+          "#text": `${options.link}${page.url}`,
+        },
+        description: page.excerpt,
```

Comment:
I think this should be configurable. Maybe something [like `metas` plugin](https://lume.land/plugins/metas/)?

Reply:
Maybe we could use the description of the meta plugin as a fallback? Or do you think it would be better if the keyword excerpt is configurable?

Reply:
I think it should be configurable. For example:

```js
site.use(rss({
  data: {
    title: "title",
    description: "excerpt",
    // etc.
  }
}));
```

Reply:
done. also changed the default for `description` to `description`